service/app/controllers/textbook/BaseTextbookController.java

```diff
@@ -0,0 +1,89 @@
+package controllers.textbook;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import controllers.BaseController;
+import org.apache.commons.io.IOUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.request.ExecutionContext;
+import org.sunbird.common.responsecode.ResponseCode;
+import play.mvc.Http;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This base controller class has all common methods for Textbook Toc API
+ * @author gauraw
+ */
+public class BaseTextbookController extends BaseController {
+
+    /**
+     * @param operation
+     * @param objectType
+     * @return
+     * @throws IOException
+     */
+    protected org.sunbird.common.request.Request createAndInitUploadRequest(
```

Comment:
Push this method to BaseController as it is reusable.

Reply:
this method is specific for textbook toc api and we may change it while doing actual implementation.

Reply:
move not reflected.